### DISCUSSION TITLE
Function was missed by many hours

### DISCUSSION THREAD

**Discussion by Originn** [agronholm/apscheduler#615] [Q&A]
I am running the following:

```
scheduler = BackgroundScheduler(timezone="Europe/London")
# Runs from Monday to Friday at 5:30 (am)
scheduler.add_job(
    func=symbol_search,
    trigger="cron",
    max_instances=1,
    day_of_week='mon-fri',
    hour=5,
    minute=30,
)
scheduler.start()
```

When I leave my bash open I see the following:

Run time of job "symbol_search (trigger: cron[day_of_week='mon-fri', hour='5', minute='30'], next run at: 2022-04-15 05:30:00 BST)" was missed by 6:10:56.555975
Run time of job "symbol_search (trigger: cron[day_of_week='mon-fri', hour='5', minute='30'], next run at: 2022-04-15 05:30:00 BST)" was missed by 6:10:57.416156

I am running this on a flask app localy and I have also a flask app on Heruko, but I don't get why it's missing by so many hours and why I have 2 alerts? The Heruko app also should be updated but I shouldn't see the output localy, so that can't be the reason.

Thanks

**Comment by agronholm** [comment]
It seems that the scheduler was dormant during that time. This would be caused by the environment (Heroku). Are you running the scheduler in a web worker?

**Comment by lasteris** [comment]
@agronholm I think i got into most similar to this discussion problem. Apscheduler 3.11, Docker Compose, Python 3.13 (Alpine) with tzdata installed.

i am observing strange here:

```
2025-05-19 13:43:52,580 DEBUG apscheduler.scheduler: Next wakeup is due at 2025-05-20 00:00:00+03:00 (in 36967.419775 seconds)
2025-05-20 01:08:26,158 DEBUG apscheduler.scheduler: Looking for jobs to run

```

Scheduler promises to run for lookup at 00:0, but actually runs at 01:08.
And then, ofc, tasks, which should run in between are logging as missed by.
But tasks, which assigned to run after 01:08 - runs successfully.

[log.txt](https://github.com/user-attachments/files/20344326/log.txt)

TZ=Europe/Moscow

**Reply by agronholm** [reply]
Can you reproduce this problem? Otherwise this is not really actionable.

**Reply by lasteris** [reply]
> Can you reproduce this problem? Otherwise this is not really actionable.

Oh...well, I will try
